pkg/cloudprovider/aws/fleet/instance.go

```diff
@@ -78,3 +81,30 @@ func (p *InstanceProvider) Create(ctx context.Context,
 	}
 	return createFleetOutput.Instances[0].InstanceIds[0], nil
 }
+
+func (p *InstanceProvider) getInstanceIDs(nodes []*v1.Node) []*string {
+	ids := []*string{}
+	for _, node := range nodes {
+		id := strings.Split(node.Spec.ProviderID, "/")
+		if len(id) < 5 {
+			zap.S().Debugf("Continuing after failure to parse instance id, %s has invalid format", node.Name)
+			continue
+		}
+		ids = append(ids, aws.String(id[4]))
+	}
+	return ids
+}
+
+func (p *InstanceProvider) TerminateInstances(ctx context.Context, nodes []*v1.Node) (*int, error) {
```

Comment:
This function name is inconsistent with other public APIs. You've named this TerminateInstances, while the other method is simply Create.
```
instanceProvider.Create()
instanceProvider.Terminate()
```
vs
```
instanceProvider.CreateInstance()
instanceProvider.TerminateInstances() 
```
IMO, the former stutters less.

Reply:
Got it. I agree.